cmd/sysinfo/sysinfo.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2021 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sysinfo
+
+import (
+	"runtime"
+
+	"github.com/spf13/cobra"
+	system "k8s.io/system-validators/validators"
+)
+
+func NewSysinfoCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "sysinfo",
+		Short: "Display system information",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runSysinfo()
+		},
+	}
+
+}
+
+func runSysinfo() error {
+	reporter := system.DefaultReporter
+
+	var validators = []system.Validator{
+		&system.KernelValidator{Reporter: reporter},
+	}
+
+	if runtime.GOOS == "linux" {
+		validators = append(validators,
+			&system.OSValidator{Reporter: reporter},
+			&system.CgroupsValidator{Reporter: reporter},
+		)
+	}
+
+	spec := system.DefaultSysSpec
+	spec.KernelSpec.Required =
+		[]system.KernelConfig{
+			{Name: "NAMESPACES"},
+			{Name: "NET_NS"},
+			{Name: "PID_NS"},
+			{Name: "IPC_NS"},
+			{Name: "UTS_NS"},
+			{Name: "CGROUPS"},
+			{Name: "CGROUP_CPUACCT"},
+			{Name: "CGROUP_DEVICE"},
+			{Name: "CGROUP_FREEZER"},
+			{Name: "CGROUP_PIDS"},
+			{Name: "CGROUP_SCHED"},
+			{Name: "CPUSETS"},
+			{Name: "MEMCG"},
+			{Name: "INET"},
+			{Name: "EXT4_FS"},
+			{Name: "PROC_FS"},
+			{Name: "NETFILTER_XT_TARGET_REDIRECT", Aliases: []string{"IP_NF_TARGET_REDIRECT"}},
+			{Name: "NETFILTER_XT_MATCH_COMMENT"},
+			{Name: "FAIR_GROUP_SCHED"},
+		}
+
+	spec.KernelSpec.Optional =
+		[]system.KernelConfig{
+			{Name: "OVERLAY_FS", Aliases: []string{"OVERLAYFS_FS"}, Description: "Required for overlayfs."},
+			{Name: "BLK_DEV_DM", Description: "Required for devicemapper."},
+			{Name: "CFS_BANDWIDTH", Description: "Required for CPU quota."},
+			{Name: "CGROUP_HUGETLB", Description: "Required for hugetlb cgroup."},
+			{Name: "SECCOMP", Description: "Required for seccomp."},
+			{Name: "SECCOMP_FILTER", Description: "Required for seccomp mode 2."},
+
```

Comment:
extra line break

Reply:
The configs above the whitespace is from https://github.com/kubernetes/system-validators/blob/master/validators/types_unix.go#L54 and have description, while the ones below are based on our own tests and sorted by name.

Reply:
ok makes sense. i thought it was a typo